commitlint_range.sh

```diff
@@ -6,12 +6,26 @@ set -o errexit
 # set -o xtrace
 # set -o pipefail
 
-if [ -n "$CIRCLE_PULL_REQUEST" ]
+
+
+
+if [ -n "${CIRCLE_PULL_REQUEST}" ]
 then
   echo "This is a pull request"
   PULL_REQUEST_NUMBER=$(echo "${CIRCLE_PULL_REQUEST}" | cut -d/ -f7)
-  BASE_SHA1=$(curl https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${PULL_REQUEST_NUMBER} | jq -r .base.sha)
-  HEAD_SHA1=$(curl https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${PULL_REQUEST_NUMBER} | jq -r .head.sha)
+  PULL_REQUEST_DETAILS=$(curl https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${PULL_REQUEST_NUMBER})
+  REPO_IS_PRIVATE=$(echo "${PULL_REQUEST_DETAILS}" | jq -r .url)
+  if [ -n "${REPO_IS_PRIVATE}" ]
```

Comment:
Unless I am misunderstanding this script, the logic is backwards.

It needs to be:

```
...
if [ -n "${REPO_IS_PRIVATE}" ]
  then
    echo "Public repo"
  else
    echo "Private repo"
    PULL_REQUEST_DETAILS=$(curl -H "Authorization: token ${GITHUB_TOKEN_COMMITLINT}" https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${PULL_REQUEST_NUMBER})
fi
...
```

See also: https://stackoverflow.com/a/3601734